examples/multimodal/components/decode_worker.py

```diff
@@ -141,25 +142,42 @@ async def async_init(self):
                 vision_tower.vision_model.embeddings.position_embedding.num_embeddings
             )
         else:
-            enc_comp_ns, enc_comp_name = EncodeWorker.dynamo_address()  # type: ignore
+            EMBEDDINGS_SHAPE = (1, 577, 4096)
```

Comment:
Instead of hardcoded here, could we use something like line 142 to get the shape
```
self.embedding_size = (
                vision_tower.vision_model.embeddings.position_embedding.num_embeddings
            )
```

Reply:
makes sense, once I get my test machine allocation back I'll make and verify the change.

Reply:
`vision_tower` is not defined in the scope; so it doesn't seem like this will work.

Reply:
Maybe we could make the vision_tower one layer before so that it could be referred to both if-else section?

Reply:
but the model doesn't need to be loaded in both cases, right? isn't that just wasting GPU resources?

Reply:
That's a good concern, maybe we could unload the model? Just wanted to see if we could get rid of the hardcoded stuff and infer that info somehow.

Reply:
Maybe can fix this as follow-up in a later phase, not a blocker for this PR. Could we add TODO comment for it?